### PR TITLE
TAR-666: Add version pinning for docker-ce-cli

### PR DESCRIPTION
**Pull request by seemethere**
Tested with the following versions:

```bash
#!/usr/bin/env bash

set -e

RUN="docker run --rm -it -v $(pwd):/v -w /v"

for distro in ubuntu:bionic debian:stretch centos:7 fedora:29; do
        for version in 18.09.2 18.06.1; do
                echo "${distro}, VERSION: ${version}"
                ${RUN} "${distro}" sh -c "VERSION=${version} sh install.sh && docker --version"
        done
done
```

### CODE REVIEW

**Comment by seemethere** [comment]
@thaJeztah Since containerd.io doesn't necessarily correspond to a specific release of the engine/cli I'm going to not include it in the script. (it's too much business logic to maintain without adding extra env variables to tune)

**Comment by thaJeztah** [comment]
Yes, not pinned to a version, more "make sure it's updated if we run the script"

**Comment by seemethere** [comment]
@thaJeztah we could have the script run an upgrade for containerd.io if it's already installed but then again I think the overlap for people who already have containerd.io installed and would use this script to install docker-ce is probably pretty low.

File: install.sh [hunk 1]
```diff
@@ -437,6 +446,10 @@ do_install() {
 				if ! is_dry_run; then
 					set -x
 				fi
+				# install the correct cli version first
+				if [ -n "$cli_pkg_version" ]; then
+					$sh_c "$pkg_manager install -y -q docker-ce-cli-$cli_pkg_version"
```
Comment (StefanScherer) [inline install.sh]: @seemethere a = instead of - before $cli_pkg_version ?

Comment (seemethere) [reply install.sh]: I'm just checking here if it's non-empty

Comment (thaJeztah) [reply install.sh]: I think he meant;

```suggestion
					$sh_c "$pkg_manager install -y -q docker-ce-cli=$cli_pkg_version"
```

Comment (seemethere) [reply install.sh]: Oh for apt it's an equals sign not a dash

Comment (thaJeztah) [reply install.sh]: Yes, I missed it as well (good 👀 @StefanScherer 🤗)

Comment (seemethere) [reply install.sh]: Wait sorry, lost track of where we were in the code, a `-` is the right character for `yum/dnf` install

Comment (thaJeztah) [reply install.sh]: 🤦‍♂️ ah, you're right. Perhaps we should change `$pkg_manager` into `$yum` or `$yum_or_dnf` 😅

Comment (StefanScherer) [reply install.sh]: Ah ok, didn't saw that from the phone that this is in the centos case. 🔍

**Pull request merged**